logger/mmap_buffer.cc

```diff
@@ -112,12 +112,9 @@ TfwMmapBufferReader::read()
 
 	const char *data = buf_->data + (tail & buf_->mask);
 	const uint64_t size = head - tail;
-	if (size == 0) [[unlikely]]
```

Comment:
I can't understand how if fixes the issue? How I see it: if `size == 0` in the callback we don't reach `while` loop and we only call `clickhouse.commit()`. Therefore it looks like we didn't call `commit` previously and fix the problem calling `commit()` few times.